docs/pages/modules/shared-objects.mdx

```diff
@@ -0,0 +1,291 @@
+    val matrix = Matrix().apply { postRotate(degrees) }
+    current = Bitmap.createBitmap(current, 0, 0, current.width, current.height, matrix, true)
+  }
+
+  fun render(): ImageRef {
+    return ImageRef(current.copy(current.config, true), runtimeContext)
+  }
+
+  override fun sharedObjectDidRelease() {
+    if (!current.isRecycled) current.recycle()
+  }
+}
+```
+
+The above example is quite similar to the iOS implementation. However, there is one difference on Android: the `sharedObjectDidRelease()` method. This lifecycle callback is invoked when JavaScript releases all references to the shared object, providing an opportunity to clean up native resources.
+
+When the result of this class is passed to another module, the `render` method returns an `ImageRef`, which is a specialized `SharedRef<Bitmap>` type that `expo-image` and other image-aware modules already understand.
+
+The module definition exposes an async function to create the context and a class definition to bind methods. The Expo Modules API uses a declarative syntax where you specify the module name, functions to create instances, and a class definition that maps methods to the shared object:
+
+```kotlin
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
+
+class SimpleImageModule : Module() {
+  override fun definition() = ModuleDefinition {
+    Name("SimpleImageModule")
+
+    AsyncFunction("createContextAsync") { path: String ->
+      val bitmap = BitmapFactory.decodeFile(path)
+        ?: throw Exceptions.IllegalArgument("Unable to decode image at $path")
+      SimpleImageContext(runtimeContext, bitmap)
+    }
+
+    Class<SimpleImageContext>("Context") {
+      Function("rotate") { ctx: SimpleImageContext, degrees: Float -> ctx.rotate(degrees) }
+      AsyncFunction("renderAsync") Coroutine { ctx: SimpleImageContext -> ctx.render() }
+      Function("reset") { ctx: SimpleImageContext -> ctx.apply { rotate(0f) } }
+    }
+  }
+}
+```
+
+In the above example, `createContextAsync` function decodes the bitmap from the file path and returns a new `SimpleImageContext` instance. Once the context exists, the `rotate` function runs synchronously because it only manipulates in memory. The `renderAsync` function is marked async to signal that it might involve copying or preparing the bitmap for consumption by other modules.
+
+### iOS implementation
+
+In iOS, you create a shared object by inheriting from `SharedObject` class provided by `ExpoModulesCore`. This class manages the decoded bitmap and exposes methods to manipulate it. The implementation keeps track of both the original image and the current state after any transformation, allowing you to reset to the original image if needed:
+
+```swift
+import ExpoModulesCore
+import UIKit
+
+final class ImageRef: SharedRef<UIImage> {
+  init(_ image: UIImage) {
+    super.init(object: image)
+  }
+}
```

Comment:
We don't need the initializer, `SharedRef` already has one that stores the argument as `ref` property: https://github.com/expo/expo/blob/12b1bbc690d2a6038e7f96ccd46b4330b9d3278c/packages/expo-modules-core/ios/Core/SharedObjects/SharedRef.swift#L24-L27

```suggestion
final class ImageRef: SharedRef<UIImage> {}
```